src/RouteCollector.php

```diff
@@ -49,9 +49,15 @@ class RouteCollector
      */
     private $routes = [];
 
-    public function __construct(RouterInterface $router)
+    /**
+     * @var DuplicateRouteDetector
+     */
+    private $duplicateRouteDetector;
+
+    public function __construct(RouterInterface $router, DuplicateRouteDetector $duplicateRouteDetector)
     {
         $this->router = $router;
+        $this->duplicateRouteDetector = $duplicateRouteDetector;
     }
```

Comment:
```suggestion
    public function __construct(RouterInterface $router, ?DuplicateRouteDetector $duplicateRouteDetector = null)
    {
        $this->router = $router;
        $this->duplicateRouteDetector = $duplicateRouteDetector ?: new DuplicateRouteDectector();
    }
```

Since it has no constructor arguments, making it nullable and providing a default will keep BC for end-users.